modality-probe-cli/src/visualize/graph.rs

```diff
@@ -219,13 +219,26 @@ fn graph_to_tree<'a>(
         });
 
         if let Ok(emeta) = meta::get_event_meta(cfg, &node.probe_id, &node.id) {
-            let payload =
-                meta::parsed_payload(emeta.type_hint.as_ref().map(|s| s.as_ref()), node.payload)
-                    .unwrap_or(None);
+            let payload = node.payload.and_then(|pl| {
+                emeta.type_hint.as_ref().and_then(|th| {
+                    meta::parsed_payload(Some(th.as_ref()), Some(pl))
+                        .ok()
+                        .flatten()
+                })
+            });
+            let has_log_str = emeta.description.contains("{}");
+            let log_str = if has_log_str && payload.is_some() {
+                Some(emeta.description.replace("{}", payload.as_ref().unwrap()))
```

Comment:
Maybe this should be using the `format_payload()` method from the `DescriptionFormat` trait?

Reply:
Yep! just made this change. Pushing shortly.